File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/domain/DeleteRecipeUseCaseTest.java
package com.tarbadev.witchcraft.domain;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;
import org.springframework.test.context.ActiveProfiles;

import static org.mockito.Mockito.verify;

@RunWith(MockitoJUnitRunner.class)
@ActiveProfiles("test")
public class DeleteRecipeUseCaseTest {
  @Mock private RecipeRepository recipeRepository;

  private DeleteRecipeUseCase subject;

  @Before
  public void setUp() {
    subject = new DeleteRecipeUseCase(recipeRepository);
  }

  @Test
  public void execute() {
    subject.execute(123);

    verify(recipeRepository).delete(123);
  }
}

File: WitchcraftApp/src/test/kotlin/com/tarbadev/witchcraft/recipes/domain/usecase/DeleteRecipeUseCaseTest.kt
package com.tarbadev.witchcraft.recipes.domain.usecase

import com.nhaarman.mockitokotlin2.mock
import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.Mockito.verify
import org.springframework.test.context.ActiveProfiles

@ActiveProfiles("test")
class DeleteRecipeUseCaseTest {
    private val recipeRepository: RecipeRepository = mock()

    private lateinit var deleteRecipeUseCase: DeleteRecipeUseCase

    @BeforeEach
    fun setUp() {
        deleteRecipeUseCase = DeleteRecipeUseCase(recipeRepository)
    }

    @Test
    fun execute() {
        deleteRecipeUseCase.execute(123)

        verify<RecipeRepository>(recipeRepository).delete(123)
    }
}